tests/Integration/Integration.Tests/LedgerPeriodModuleTests/ContextBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace Linn.Purchasing.Integration.Tests.LedgerPeriodModuleTests
+{
+    using System.Net.Http;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Logging;
+    using Linn.Common.Persistence;
+    using Linn.Purchasing.Domain.LinnApps;
+    using Linn.Purchasing.IoC;
+    using Linn.Purchasing.Resources;
+    using Linn.Purchasing.Service.Modules;
+
+    using Microsoft.Extensions.DependencyInjection;
+
+    using NSubstitute;
+
+    using NUnit.Framework;
+
+    public class ContextBase
+    {
+        protected HttpClient Client { get; set; }
+
+        protected HttpResponseMessage Response { get; set; }
+
+        protected ITransactionManager TransactionManager { get; set; }
+
+        protected IFacadeResourceService<LedgerPeriod, int, LedgerPeriodResource, LedgerPeriodResource> LedgerPeriodFacadeService { get; private set; }
+
+        [SetUp]
+        public void EstablishContext()
+        {
+            this.TransactionManager = Substitute.For<ITransactionManager>();
+
+            this.LedgerPeriodFacadeService = Substitute.For<IFacadeResourceService<LedgerPeriod, int, LedgerPeriodResource, LedgerPeriodResource>>();
```

Comment:
I prefer using a real FacadeService in my integration tests...

 although I am wary that by introducing different testing styles I am pushing things to be less standardised

Reply:
yeah. you're right. there's no reason I shouldn't use a real one in simple examples like these. I'll think about it next time.